en/guide/create_message.md

```diff
@@ -0,0 +1,246 @@
+As a concrete example, the definition of the `BATTERY_STATUS` message is given below (as defined **mavlink/message_definitions/v1.0/common.xml**).
+
+> **Note** This message was chosen as it contains many of the main fields and attributes. 
+
+```xml
+    <message id="147" name="BATTERY_STATUS">
+      <description>Battery information</description>
+      <field type="uint8_t" name="id">Battery ID</field>
+      <field type="uint8_t" name="battery_function" enum="MAV_BATTERY_FUNCTION">Function of the battery</field>
+      <field type="uint8_t" name="type" enum="MAV_BATTERY_TYPE">Type (chemistry) of the battery</field>
+      <field type="int16_t" name="temperature" units="cdegC">Temperature of the battery. INT16_MAX for unknown temperature.</field>
+      <field type="uint16_t[10]" name="voltages" units="mV">Battery voltage of cells. Cells above the valid cell count for this battery should have the UINT16_MAX value.</field>
+      <field type="int16_t" name="current_battery" units="cA">Battery current, -1: autopilot does not measure the current</field>
+      <field type="int32_t" name="current_consumed" units="mAh">Consumed charge, -1: autopilot does not provide consumption estimate</field>
+      <field type="int32_t" name="energy_consumed" units="hJ">Consumed energy, -1: autopilot does not provide energy consumption estimate</field>
+      <field type="int8_t" name="battery_remaining" units="%">Remaining battery energy. Values: [0-100], -1: autopilot does not estimate the remaining battery.</field>
+      <extensions/>
+      <field type="int32_t" name="time_remaining" units="s">Remaining battery time, 0: autopilot does not provide remaining battery time estimate</field>
+      <field type="uint8_t" name="charge_state" enum="MAV_BATTERY_CHARGE_STATE">State for extent of discharge, provided by autopilot for warning or external reactions</field>
+    </message>
+```
+    
+
+The main message tags/fields are:
+
+- `message`: Each message is encapsulated by `message` tags, with the following attributes
+  - `id`: The id attribute is the unique index number of this message (in this case 147). 
+    - For MAVLink 1:
+      - Valid numbers range from 0 to 255.
+      - The ids 0-149 and 230-255 are reserved for *common.xml*. Dialects can use 150-229 (?240) for custom messages (provided these are not used by other included dialects). 
+    - For MAVLink 2:
+      - Valid numbers range from 0 to 16777215.
+      - All numbers below 255 should be considered reserved unless messages are also intended for MAVLink 1. 
+        > **Note** IDs are precious in MAVLink 1!
+  - `name`: The name attribute provides a human readable form for the message (ie "BATTERY_STATUS"). It is used for naming helper functions in generated libraries, but is not sent over the wire.
+- `description` (optional): Human readable description of message, shown in user interfaces and in code comments. 
```

Comment:
I'd vote that we should change this field to *NOT* be optional!

Reply:
@amilcarlucas I agree. I created a PR for discussion of this. https://github.com/ArduPilot/pymavlink/pull/215